gpt_index/vector_stores/deeplake.py

```diff
@@ -217,13 +226,24 @@ def delete(self, doc_id: str, **delete_kwargs: Any) -> None:
             self.ds.commit(f"deleted {len(ids)} samples", allow_empty=True)
             self.ds.summary()
 
-    def query(self, query: VectorStoreQuery) -> VectorStoreQueryResult:
+    def query(
+        self, query: VectorStoreQuery, filter: Optional[str] = None
+    ) -> VectorStoreQueryResult:
```

Comment:
We can't really change the method signature like this since `VectorStore` interface expects just the query param

Reply:
I see ... But does it break anything? If yes, I will remove it.  On the other hand we can pass **kwargs parameter for these kind of use cases.